File: my_time.py
class Time():
    def __init__(self, a, b):
        self.a = a
        self.b = b
        self.single_numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] 

    def validate_object(self):
        if self.a > 23:
            self.a = 0
        if self.b > 59:
            rermains = self.b % 60
            division_result = self.b / 60
            self.a += int(division_result)
            self.b = 0 + rermains

    def add_hour(self, hours):
        self.a += hours

    def add_minute(self, minutes):
        self.b += minutes

    def get_time(self):
        self.validate_object()
        if self.a in self.single_numbers:
            self.a_disp = f'0{self.a}'
        else: 
            self.a_disp = self.a
        if self.b in self.single_numbers:
            self.b_disp = f'0{self.b}'
        else:
            self.b_disp = self.b
        return f'{self.a_disp}:{self.b_disp}'

my_time = Time(9, 0)

print(my_time.get_time())
my_time.add_minute(43)
my_time.add_minute(20)
my_time.add_hour(14)
my_time.add_minute(57)
my_time.add_minute(5)
print(my_time.get_time())

File: my_time.js
class Time {
    constructor(a, b, singleNumbers) {
        this.a = a;
        this.b = b;
        this.singleNumbers = singleNumbers;
    }

    validateObject() {
        if (this.b > 59) {
            let remains_result = this.b % 60;
            let division_result = this.b / 60;
            this.a += parseInt(division_result);
            this.b = 0 + remains_result;
        }
        if (this.a > 23) {
            this.a = 0;
        }
    }

    addHour(hours) {
        this.a += hours
    }

    addMinute(minutes) {
        this.b += minutes
    }

    getTime() {
        let a_disp = NaN;
        let b_disp = NaN;
        this.validateObject()
        if (this.a in this.singleNumbers) {
            a_disp = `0${this.a}`;
        }
        else {
            a_disp = this.a;
        }
        if (this.b in this.singleNumbers) {
            b_disp = `0${this.b}`;
        }
        else {
            b_disp = this.b;
        }
        return `${a_disp}:${b_disp}`;
    }
}
const singleNumbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
const myTime = new Time(9, 0, singleNumbers);

// console.log(myTime.getTime())
myTime.addMinute(43)
myTime.addMinute(20)
myTime.addHour(14)
myTime.addMinute(57)
myTime.addMinute(5)
console.log(myTime.getTime())
